hoomd/DomainDecomposition.cc

```diff
@@ -354,8 +358,8 @@ bool DomainDecomposition::findDecomposition(unsigned int nranks,
 
     // initial guess
     nx = 1;
-    ny = 1;
-    nz = nranks;
+    ny = nranks;
+    nz = 1;
```

Comment:
Make the initial guess conditional on `is2D` to keep the existing behavior for 3D boxes where the z direction is divided first.